src/main/java/br/com/senior/core/base/RequestUtils.java

```diff
@@ -119,14 +118,13 @@ private HttpResponse executePost(String url, Object payload, CloseableHttpClient
     private <T> T readResponse(HttpResponse response, Class<T> clazz) throws IOException, ServiceException {
         int statusCode = response.getStatusLine().getStatusCode();
         if (statusCode >= 400) {
-            ErrorOutput errorOutput = new Gson().fromJson(new InputStreamReader(response.getEntity().getContent(), StandardCharsets.ISO_8859_1), ErrorOutput.class);
-            log.error("Erro ao efetuar requisição, código de erro: {}. Erro retornado: {}", statusCode, errorOutput);
-            throw new ServiceException(statusCode, errorOutput.getMessage());
+            String body = EntityUtils.toString(response.getEntity());
+            throw new ServiceException(statusCode, body);
         }
         if (statusCode == 204 && response.getEntity() == null) {
             return null;
         }
-        return new Gson().fromJson(new InputStreamReader(response.getEntity().getContent(), StandardCharsets.ISO_8859_1), clazz);
+        return SeniorGsonBuilder.newGsonBuilder().fromJson(new InputStreamReader(response.getEntity().getContent(), StandardCharsets.ISO_8859_1), clazz);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *InputStreamReader*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by Gson.fromJson. Either a) close the object returned by *InputStreamReader()* in a try-finally block or b) close the resource by declaring the object returned by *InputStreamReader()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).